include/swift/AST/Types.h

```diff
@@ -2295,7 +2323,8 @@ getSILFunctionLanguage(SILFunctionTypeRepresentation rep) {
 /// be 'thin', indicating that a function value has no capture context and can be
 /// represented at the binary level as a single function pointer.
 class AnyFunctionType : public TypeBase {
-  const Type Input;
+  const ArrayRef<CallArgParam> Input;
```

Comment:
`llvm::TrailingObjects` requires this class be `final`.  I can manually tail-allocate if you still want it.